src/sentry/static/sentry/app/components/smartSearchBar/index.tsx

```diff
@@ -1336,14 +1350,14 @@ const InputButton = styled(Button)`
 const StyledDropdownLink = styled(DropdownLink)`
   display: none;
 
-  @media (max-width: ${p => p.theme.breakpoints[2]}) {
+  @media (max-width: ${commonTheme.breakpoints[2]}) {
     display: flex;
   }
 `;
 
-const DropdownElement = styled('a')`
+const DropdownElement = withTheme(styled('a')`
```

Comment:
are you sure about this? you should be able to access the theme in a  styled component

Reply:
Absolutely, but if I don't put it, I get `TS2741: Property 'theme' is missing in type '{ children: (string | Element)[]; showBelowMediaQuery: number; "data-test-id": string; onClick: (evt: MouseEvent<Element, MouseEvent>) => Promise<void>; }' but required in type '{ theme: Theme; showBelowMediaQuery: number; last?: boolean | undefined; }'.` which I was unable to figure out how to fix. It's something about passing function components to `styled` template string parameter. Like the previous one, I'd be happy if someone could explain this to me.

Reply:
The issue is that when we render `DropdownElement` here https://github.com/getsentry/sentry/blob/6225cea4e273b49985ac9458289a2c9ef6061630/src/sentry/static/sentry/app/components/smartSearchBar/index.tsx#L1257-L1260 - it expects `theme` because of https://github.com/getsentry/sentry/blob/6225cea4e273b49985ac9458289a2c9ef6061630/src/sentry/static/sentry/app/components/smartSearchBar/index.tsx#L84-L88

Which is necessary because we use it outside of a styled component... I'm not sure if there's a cleaner way to do this but you can extract the param type of `getDropdownElementStyles`, and type `DropdownElement` with it - we omit `theme`, because `styled()` injects it.

```
const DropdownElement =styled('a')<Omit<DropdownElementStylesProps, 'theme'>>`
```

Reply:
I'm not sure what you're suggesting here. Are you saying to extract the getDropdownElementStyles props to its own type, and then pass it here? It seems like a lot of boilerplate, but I guess it does avoid adding another React wrapper.

Reply:
Yes, exactly, it's what we do in other components.

Reply:
OK. I don't really like the complicated code, but I think the small runtime benefit is worth the small code complexity here, so I'll fix this.